src/compose/app.ts

```diff
@@ -336,15 +336,10 @@ export class App {
 				return false;
 			}
 
-			// Check if we previously remember starting it
-			if (
-				applicationManager.containerStarted[serviceCurrent.containerId!] != null
-			) {
-				return false;
-			}
-
-			// If the config otherwise matches, then we should be running
-			return isEqualExceptForRunningState(serviceCurrent, serviceTarget);
+			// Only start a Service if we have never started it before!
+			// Otherwise, all other states regarding starting will be handled by the engine.
+			// This is so the engine can enforce the restart policy configured for the container.
+			return serviceCurrent.status === 'Installed';
```

Comment:
Service changes count as an install, correct? Like changing env vars or updating images?

What about service restarts via the dashboard, does that count as an install?

I'm sure the answer is obvious if I dug through the code but it might be helpful to mention here as a comment for context.

Reply:
Yeah, all of those operations recreate the container so they have an intermediate `Install` step @klutchell 

I wonder if this should include `Dead` containers, not sure if the engine restarts those.

Reply:
`Dead` containers cannot be started again https://github.com/docker/cli/issues/502#issuecomment-330361748

> What about service restarts via the dashboard, does that count as an install?

It's possible this change might break that so I'm doing some testing specifically for that situation